### PR TITLE
Initial upload

### PR DESCRIPTION
**Pull request by artivis**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,80 @@
...
+Note that when launching the `joy` or `key` app,
+the appropriate topic is automatically selected.
+
+Note that this snap is part of an integrated snaps deployment.
+Other recommended snaps to be installed are,
+
+- [micro-xrce-dds-agent](LINK)
+- [sllidar-ros2](https://snapcraft.io/sllidar-ros2)
+- [rosbot-xl](https://snapcraft.io/rosbot-xl)
+- [rosbot-xl-nav](https://snapcraft.io/rosbot-xl-nav)
```
Comment (Kotochleb) [inline README.md]:
```suggestion
> **Note**
> This snap is part of an integrated snaps deployment.
> 
> Other recommended snaps to be installed are,
> 
> - [micro-xrce-dds-agent](LINK)
> - [sllidar-ros2](https://snapcraft.io/sllidar-ros2)
> - [rosbot-xl-teleop](https://snapcraft.io/rosbot-xl-teleop)
> - [rosbot-xl-nav](https://snapcraft.io/rosbot-xl-nav)
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,80 @@
...
+network              rosbot-xl:network       :network                        -
+network-bind         rosbot-xl:network-bind  :network-bind                   -
+```
+
+The interface `ros-humble` must be connected.
+
+If it isn't, you can issue the following command,
+
+```bash
+snap connect rosbot-xl:ros-humble ros-humble-ros-base
```
Comment (Kotochleb) [inline README.md]:
```suggestion
snap connect rosbot-xl-teleop:ros-humble ros-humble-ros-base
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,80 @@
...
+Upon installation, depending on your operating system,
+you may have to manually connect the snap interface.
+You can verify that with the following command,
+
+```bash
+$ snap connections rosbot-xl
+Interface            Plug                    Slot                            Notes
+content[ros-humble]  rosbot-xl:ros-humble    ros-humble-ros-base:ros-humble  manual
+network              rosbot-xl:network       :network                        -
+network-bind         rosbot-xl:network-bind  :network-bind                   -
```
Comment (Kotochleb) [inline README.md]:
```suggestion
$ snap connections rosbot-xl-teleop
Interface            Plug                               Slot                            Notes
content[ros-humble]  rosbot-xl-teleop:ros-humble        ros-humble-ros-base:ros-humble  manual
hardware-observe     rosbot-xl-teleop:hardware-observe  -                               -
joystick             rosbot-xl-teleop:joystick          -                               -
network              rosbot-xl-teleop:network           :network                        -
network-bind         rosbot-xl-teleop:network-bind      :network-bind                   -

```

File: snap/local/mux_completer.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/bash -e
+
+complete -W "joy_vel key_vel nav_vel web_vel" rosbot-xl-teleop.mux
```
Comment (Kotochleb) [inline snap/local/mux_completer.sh]:
```suggestion
complete -W "joy_vel key_vel nav_vel web_vel" rosbot-xl-teleop.mux-select
```

File: snap/local/reset_config.sh [hunk 5]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh -e
+
+# Make sure the folder exists
+mkdir -p $SNAP_COMMON/config
+
+cp $SNAP/usr/share/rosbot-xl-teleop/config/*.yaml $SNAP_COMMON/config/.
```
Comment (Kotochleb) [inline snap/local/reset_config.sh:6]:
```suggestion
cp $SNAP/usr/share/rosbot-xl-teleop/config/*.yaml $SNAP_COMMON/config/
```
Is that dot necessary?

**Review by Kotochleb — CHANGES_REQUESTED** [review]
So far it mostly looked ok and worked. Yet I was unable to test the joystick

File: snap/local/reset_config.sh (hunk 5 above)
Comment (artivis) [reply snap/local/reset_config.sh:6]: It is not. Just a convention to mean "this directory".

File: README.md [hunk 6]
```diff
@@ -1,2 +1,90 @@
...
+a gamepad or leave it to the navigation stack.
+
+[![Get it from the Snap Store](https://snapcraft.io/static/images/badges/en/snap-store-black.svg)](https://snapcraft.io/rosbot-xl-teleop)
+
+## Installation
+
+Install the snap as follows,
+
+```bash
+snap install rosbot-xl-teleop
```
Comment (Guillaumebeuzeboc) [inline README.md:13]: sudo is required and so far nothing is uploaded on `latest/stable`.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,90 @@
...
+network-bind         rosbot-xl-teleop:network-bind      :network-bind                   -
+
+```
+
+The interface `ros-humble` must be connected.
+
+If it isn't, you can issue the following command,
+
+```bash
+snap connect rosbot-xl-teleop:ros-humble ros-humble-ros-base
```
Comment (Guillaumebeuzeboc) [inline README.md:38]: maybe the instruction to connect `hardware-observe` would be useful.

Comment (Kotochleb) [reply README.md:38]: I second this

Comment (artivis) [reply README.md:38]: Done.

File: snap/local/joy_launcher.sh [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+if [[ -z "${ROSBOT_XL_TELEOP_JOY_PARAM_FILE}" ]]; then
+  ROSBOT_XL_TELEOP_JOY_PARAM_FILE="$SNAP_COMMON/config/joy_teleop.config.yaml"
+fi
+
+# @todo(artivis) This actually has no effect and must be fixed in the teleop package
+if [[ -z "${ROSBOT_XL_TELEOP_JOY_DEV}" ]]; then
+  ROSBOT_XL_TELEOP_JOY_DEV="/dev/input/js0"
+fi
+
+$SNAP/ros2 launch teleop_twist_joy teleop-launch.py \
```
Comment (Kotochleb) [inline snap/local/joy_launcher.sh]:
```suggestion
ros2 launch teleop_twist_joy teleop-launch.py \
```

**Review by Kotochleb — CHANGES_REQUESTED** [review]
One last change and we are ready to merge

**Review by Kotochleb — APPROVED** [review]
LGTM

**Pull request merged**